type script/unAuthorized ~@Error.ts

```diff
@@ -0,0 +1,28 @@
+import { Inject, Optional } from "@angular/core";
+import { LocalStorageService } from "@core/services/local-storage.service";
+import { ModalService } from "src/app/shared/services/modal.service";
+import { ModalConstants } from "src/app/utils/modals-constants";
+
+export class UnAuthorizedError {
+
+    constructor(@Optional() @Inject(ModalService) private _modalService: ModalService,
+    @Optional() @Inject(LocalStorageService) private _localStorageService: LocalStorageService) {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider required dependencies or add checks</b></div>
<div id="fix">

Consider making the dependencies required instead of optional since `_modalService` and `_localStorageService` are used in `handleUnauthorizedError` without null checks. If these services are truly optional, null checks should be added before using them.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -8,20 +8,24 @@
     constructor(@Inject(ModalService) private _modalService: ModalService,
         @Inject(LocalStorageService) private _localStorageService: LocalStorageService) {
         this.handleUnauthorizedError();
     }

     async handleUnauthorizedError() {
         console.log('inside handle unauthorized error');
         if (!this._modalService || !this._localStorageService) {
             return;
         }

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/24#issuecomment-2514645264>#10829c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged